containers/ecr-viewer/src/app/api/save-fhir-data/save-fhir-data-service.ts

```diff
@@ -171,266 +172,158 @@ const saveFhirMetadata = async (
-            .input("city", sql.VarChar(255), address.city)
-            .input("district", sql.VarChar(255), address.district)
-            .input("state", sql.VarChar(255), address.state)
-            .input("postal_code", sql.VarChar(20), address.postal_code)
-            .input("country", sql.VarChar(255), address.country)
-            .input("period_start", sql.DateTime, address.period_start)
-            .input("period_end", sql.DateTime, address.period_end)
-            .input("eICR_ID", sql.VarChar(200), ecrId)
-            .query(
-              "INSERT INTO ecr_viewer.patient_address (UUID, [use], type, text, line, city, district, state, postal_code, country, period_start, period_end, eICR_ID) VALUES (@UUID, @use, @type, @text, @line, @city, @district, @state, @postal_code, @country, @period_start, @period_end, @eICR_ID)",
-            );
+      await (db as Kysely<Extended>).transaction().execute(async (trx) => {
+        await trx
+          .insertInto("ecr_data")
+          .values({
+            eICR_ID: ecrId,
+            set_id: metadata.eicr_set_id,
+            fhir_reference_link: null, // Not implemented yet
+            last_name: metadata.last_name,
+            first_name: metadata.first_name,
+            birth_date: metadata.birth_date,
+            gender: metadata.gender,
+            birth_sex: metadata.birth_sex,
+            gender_identity: metadata.gender_identity,
+            race: metadata.race,
+            ethnicity: metadata.ethnicity,
+            latitude: metadata.latitude,
+            longitude: metadata.longitude,
+            homelessness_status: metadata.homelessness_status,
+            disabilities: metadata.disabilities,
+            tribal_affiliation: metadata.tribal_affiliation,
+            tribal_enrollment_status: metadata.tribal_enrollment_status,
+            current_job_title: metadata.current_job_title,
+            current_job_industry: metadata.current_job_industry,
+            usual_occupation: metadata.usual_occupation,
+            usual_industry: metadata.usual_industry,
+            preferred_language: metadata.preferred_language,
+            pregnancy_status: metadata.pregnancy_status,
+            rr_id: metadata.rr_id,
+            processing_status: metadata.processing_status,
+            eicr_version_number: metadata.eicr_version_number,
+            authoring_date: metadata.authoring_datetime,
+            authoring_provider: metadata.provider_id,
+            provider_id: metadata.provider_id,
+            facility_id: metadata.facility_id_number,
+            facility_name: metadata.facility_name,
+            encounter_type: metadata.encounter_type,
+            encounter_start_date: metadata.encounter_start_date,
+            encounter_end_date: metadata.encounter_end_date,
+            reason_for_visit: metadata.reason_for_visit,
+            active_problems: metadata.active_problems,
+          })
+          .executeTakeFirstOrThrow();
```

Comment:
why `executeTakeFirstOrThrow` instead of just `execute` for these given we don't really expect a result?